boards/seeed/xiao_rp2040/xiao_rp2040.dts

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) 2021 Yonatan Schachter
+ * Copyright (c) 2023 Seeed Studio inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+
+#include <rpi_pico/rp2040.dtsi>
+#include "xiao_rp2040-pinctrl.dtsi"
+#include "seeed_xiao_connector.dtsi"
+#include <freq.h>
+#include <zephyr/dt-bindings/led/led.h>
+#include <zephyr/dt-bindings/pwm/pwm.h>
+
+/ {
+	chosen {
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,flash-controller = &ssi;
+		zephyr,console = &uart0;
+		zephyr,shell-uart = &uart0;
+		zephyr,code-partition = &code_partition;
+	};
+
+	aliases {
+		watchdog0 = &wdt0;
+		led-strip = &ws2812;
+		pwm-led0 = &pwm_led0;
+		led0 = &blue_led;
+		led1 = &green_led;
+		led2 = &red_led;
+	};
+
+	pwm_leds {
+		compatible = "pwm-leds";
+		status = "disabled";
+		pwm_led0: pwm_led_0 {
+			pwms = <&pwm 9 PWM_MSEC(20) PWM_POLARITY_INVERTED>;
+			label = "PWM_LED";
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		blue_led: blue_led {
+			gpios = <&gpio0 25 GPIO_ACTIVE_LOW>;
+			label = "BLUE_LED";
+		};
+
+		green_led: green_led {
+			gpios = <&gpio0 16 GPIO_ACTIVE_LOW>;
+			label = "GREEN_LED";
+		};
+
+		red_led: red_led {
+			gpios = <&gpio0 17 GPIO_ACTIVE_LOW>;
+			label = "RED_LED";
+		};
+	};
+};
+
+&pwm {
+	pinctrl-0 = <&pwm_ch4b_default>;
+	pinctrl-names = "default";
+	divider-int-0 = <255>;
+};
+
+&flash0 {
+	/*
+	 * 2MB of flash minus the 0x100 used for
+	 * the second stage bootloader
+	 */
+	reg = <0x10000000 DT_SIZE_M(2)>;
+
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		/*
+		 * Start at the beginning of usable flash, 2MB minus the
+		 * second stage space and the 16 KiB reserved for settings
+		 */
+		code_partition: partition@100 {
+			label = "code";
+			reg = <0x100 (DT_SIZE_M(2) - 0x100)>;
+			read-only;
+		};
+	};
+};
+
+&clocks {
+	pinctrl-0 = <&clocks_default>;
+	pinctrl-names = "default";
+};
+
+&timer {
+	status = "okay";
+};
+
+&uart0 {
+	current-speed = <115200>;
+	status = "okay";
+	pinctrl-0 = <&uart0_default>;
+	pinctrl-names = "default";
+};
+
+&i2c1 {
+	status = "okay";
+	pinctrl-0 = <&i2c1_default>;
+	pinctrl-names = "default";
+	clock-frequency = <I2C_BITRATE_FAST>;
+};
+
+&spi0 {
+	status = "okay";
+	pinctrl-0 = <&spi0_default>;
+	pinctrl-names = "default";
+	clock-frequency = <DT_FREQ_M(8)>;
+};
+
+&gpio0 {
+	status = "okay";
+	neopixel-power-enable {
```

Comment:
also this seems to have been copied from `adafruit/qt_py_rp2040/adafruit_qt_py_rp2040.dts` without copying the text, can you add the text and update it?
```
/*
* Unlike some of the other Adafruit boards, the neopixel on this board has
* its positive side hooked up to a GPIO pin rather than a positive voltage
* rail to save on power. This will enable the LED on board initialization.
*/
```

Reply:
Yeah the board is very similar to the QT PY in terms of the WS2812 LED from what I can tell looking over the schematics of both (I'm a random contributor / not from Seeed so I've been using the schematic to figure it out). Looked into the RGB LED example and saw how it was accomplished there so did the same thing here. I've augmented it to add a generic message as it's not an adafruit board. Apologies.